integration_tests/plugins/test_rust_plugin.py

```diff
@@ -103,6 +107,24 @@ def test_cross_compiling(self):
         binary = os.path.join(self.parts_dir, 'rust-hello', 'install', 'bin',
                               'rust-hello')
         self.assertThat(binary, HasArchitecture('aarch64'))
+        self.assertThat(binary, HasLinkage('dynamically linked'))
```

Comment:
It seems to me that the matchers would be clearer called like IsDynamicallyLinked() and IsStaticallyLinked()

Reply:
To my mind `HasLinkage` is consistent with `HasArchitecture` in that it's not hiding the basic string-parsing nature of the matcher... but yeah, I added subclasses as per your suggestion now.